p2p/src/net/default_backend/mod.rs

```diff
@@ -142,8 +142,16 @@ where
     }
 }
 
+#[async_trait]
 impl MessagingService for MessagingHandle {
-    fn send_message(&mut self, peer_id: PeerId, message: SyncMessage) -> crate::Result<()> {
+    async fn send_message(&mut self, peer_id: PeerId, message: SyncMessage) -> crate::Result<()> {
+        // Note: send_message was made async to be able to use a bounded channel in tests.
+        // TODO: it may make sense to use a bounded channel in production as well. One possible
+        // reason for this might be to limit the number of BlockResponse messages that can exist
+        // at the same time. E.g. currently sync::Peer will produce a BlockResponse for each
+        // requested block almost immediately. The maximum number of simultaneously requested
+        // blocks is 500 per peer, a block can be up to 1Mb in size, so several malicious peers
+        // can eat up several Gb of RAM on the node.
```

Comment:
I've created an issue for it - https://github.com/mintlayer/mintlayer-core/issues/1314
As I've mentioned in the issue, I no longer think that using a bounded channel here will help, so I'll remove this TODO.